packages/tc-ui/relationship/lib/relationship-picker.jsx

```diff
@@ -0,0 +1,329 @@
+	<li
+		data-name={value.name}
+		data-code={value.code}
+		className="selected-relationship"
+	>
+		<button
+			type="button"
+			disabled={disabled ? 'disabled' : null}
+			className={`o-buttons o-buttons--small ${
+				disabled ? 'disabled' : ''
+			}`}
+			onClick={onRelationshipRemove}
+			data-index={index}
+			key={index}
+		>
+			Remove
+		</button>
+		<span className="o-layout-typography">{value.name || value.code}</span>
+	</li>
+);
+
+class RelationshipPicker extends Component {
+	constructor(props) {
+		super();
+		this.state = {
+			searchTerm: '',
+			suggestions: [],
+			isUserError: false,
+			isUnresolved: false,
+			selectedRelationships: toArray(props.value),
+			hasHighlightedSelection: false,
+		};
+		this.props = props;
+		this.onChange = this.onChange.bind(this);
+		this.onSuggestionsFetchRequested = this.onSuggestionsFetchRequested.bind(
+			this,
+		);
+		this.onSuggestionsClearRequested = this.onSuggestionsClearRequested.bind(
+			this,
+		);
+		this.onSuggestionSelected = this.onSuggestionSelected.bind(this);
+		this.onRelationshipRemove = this.onRelationshipRemove.bind(this);
+		this.onUserMisconception = this.onUserMisconception.bind(this);
+		this.onSuggestionHighlighted = this.onSuggestionHighlighted.bind(this);
+	}
+
+	onRelationshipRemove(event) {
+		this.setState(({ selectedRelationships }) => {
+			selectedRelationships = [...selectedRelationships];
+			selectedRelationships.splice(event.target.dataset.index, 1);
+			return { selectedRelationships };
+		});
+		event.preventDefault();
+	}
+
+	onChange(event, { newValue }) {
+		this.setState({
+			searchTerm: newValue,
+			isUnresolved: !!newValue,
+		});
+	}
+
+	onSuggestionsFetchRequested({ value }) {
+		if (!value) {
+			return;
+		}
+		return fetch(
+			`/autocomplete/${this.props.type}/name?q=${value}&parentType=${this.props.parentType}&propertyName=${this.props.propertyName}`,
+		)
+			.then(results => results.json())
+			.then(results => {
+				this.setState(({ selectedRelationships }) => ({
+					suggestions: results
+						// avoid new suggestions including values that have already been selected
+						.filter(
+							suggestion =>
+								!selectedRelationships.find(
+									({ code }) => code === suggestion.code,
+								),
+						),
+				}));
+			});
+	}
+
+	onSuggestionSelected(event, { suggestion }) {
+		if (this.props.hasMany) {
+			this.setState(({ selectedRelationships }) => {
+				selectedRelationships = [...selectedRelationships, suggestion];
+				return { selectedRelationships };
+			});
+		} else {
+			this.setState({
+				selectedRelationships: [suggestion],
+			});
+		}
+		this.setState({
+			searchTerm: '',
+			isUserError: false,
+			isUnresolved: false,
+		});
+		// this is needed to prevent the event propagating up  and then
+		// immediately clicking another button. VERY odd behaviour, and
+		// don't fully understand why, but this is the fix
+		if (event) {
+			event.preventDefault();
+		}
```

Comment:
I also investigated that. I'm guessing that `ReactAuthSuggest` won't care about event propagation, but I'm not sure. Anyway, this is good to fix